src/Engine/Component/Script.hpp

```diff
@@ -33,5 +34,9 @@ namespace Component {
       
             /// Entities referenced by this script.
             std::vector<Entity*> refList;
+
+            ///Map containing the properties, map<nameOfProperty, map<typeOfProperty, valueOfProperty>>
```

Comment:
Can there exist multiple properties with the same name but different types? If not, why use a map rather than a std::pair?